src/main/java/com/code_intelligence/jazzer/runtime/TraceCmpHooks.java

```diff
@@ -213,6 +213,163 @@ private static void regionMatchesInternal(
     }
   }
 
+  // For standard Kotlin packages, which are named according to the pattern kotlin.*, we append a
+  // whitespace to the package name of the target class so that they are not mangled due to shading.
+  @MethodHook(type = HookType.AFTER, targetClassName = "kotlin.jvm.internal.Intrinsics ",
+      targetMethod = "areEqual")
+  @MethodHook(
+      type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ", targetMethod = "equals")
+  @MethodHook(type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ",
+      targetMethod = "equals$default")
+  public static void
+  equalsKt(MethodHandle method, Object alwaysNull, Object[] arguments, int hookId,
+      Boolean equalStrings) {
+    if (arguments[0] instanceof String && !equalStrings) {
+      TraceDataFlowNativeCallbacks.traceStrcmp(
+          (String) arguments[0], (String) arguments[1], 1, hookId);
+    }
+  }
+
+  @MethodHook(type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ",
+      targetMethod = "contentEquals")
+  @MethodHook(type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ",
+      targetMethod = "contentEquals$default")
+  public static void
+  contentEqualKt(MethodHandle method, Object alwaysNull, Object[] arguments, int hookId,
+      Boolean equalStrings) {
+    if (arguments[0] instanceof CharSequence && !equalStrings) {
+      TraceDataFlowNativeCallbacks.traceStrcmp(((CharSequence) arguments[0]).toString(),
+          ((CharSequence) arguments[1]).toString(), 1, hookId);
+    }
+  }
+
+  @MethodHook(
+      type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ", targetMethod = "compareTo")
+  @MethodHook(type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ",
+      targetMethod = "compareTo$default")
+  public static void
+  compareToKt(
+      MethodHandle method, Object alwaysNull, Object[] arguments, int hookId, Integer returnValue) {
+    if (arguments[0] instanceof String && returnValue != 0) {
+      TraceDataFlowNativeCallbacks.traceStrcmp(
+          (String) arguments[0], (String) arguments[1], 1, hookId);
+    }
+  }
+
+  @MethodHook(
+      type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ", targetMethod = "endsWith")
+  @MethodHook(type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ",
+      targetMethod = "endsWith$default")
+  @MethodHook(type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ",
+      targetMethod = "startsWith")
+  @MethodHook(type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ",
+      targetMethod = "startsWith$default")
+  public static void
+  startsWithKt(
+      MethodHandle method, Object alwaysNull, Object[] arguments, int hookId, Boolean returnValue) {
+    if (!returnValue) {
+      TraceDataFlowNativeCallbacks.traceStrstr(
+          (String) arguments[0], (String) arguments[1], hookId);
```

Comment:
We should either check the types before casting or fix the method descriptors.

Reply:
I'm still finalizing the PR and fixing this among other issues

Reply:
@fmeum The PR can be reviewed now.